components/IscnUploadForm.vue

```diff
@@ -451,6 +456,106 @@ export default class UploadForm extends Vue {
     }
   }
 
+  async processEPub({ buffer, file }: { buffer: ArrayBuffer; file: File }) {
+    try {
+      const Book = ePub(buffer)
+      await Book.ready
+      const epubMetadata: any = {}
+
+      // Get metadata
+      const { metadata } = Book.packaging
+      if (metadata) {
+        epubMetadata.epubFileName = file.name
+        epubMetadata.title = metadata.title
+        epubMetadata.author = metadata.creator
+        epubMetadata.language = this.formatLanguage(metadata.language)
+        epubMetadata.description = metadata.description
+      }
+
+      // Get tags
+      const opfFilePath = await (Book.path as any).path
+      const opfContent = await Book.archive.getText(opfFilePath)
+      const parser = new DOMParser()
+      const opfDocument = parser.parseFromString(opfContent, 'application/xml')
+      const dcSubjectElements = opfDocument.querySelectorAll(
+        'dc\\:subject, subject',
+      )
+      const subjects: string[] = []
+      dcSubjectElements.forEach((element) => {
+        const subject = element.textContent
+        subject && subjects.push(subject)
+      })
+      epubMetadata.tags = subjects
+
+      // Get cover file
+      const coverUrl = (Book as any).cover
+      if (!coverUrl) {
+        this.epubMetadataList.push(epubMetadata)
+        return
+      }
```

Comment:
better to use `if (!coverUrl) {} else {}` or `if (coverUrl) {} else {}` instead of early return here, since people may add more code down below without realizing it is actually an if else case if we use early `return`